src/main/java/ninjabrainbot/model/input/HotkeyInputHandler.java

```diff
@@ -65,8 +67,19 @@ private void toggleAltStdIfNotLocked() {
 	}
 
 	private void toggleEnteringBoatIfNotLocked() {
-		if (preferences.usePreciseAngle.get() && !dataState.locked().get() && !dataState.allAdvancementsDataState().allAdvancementsModeEnabled().get())
+		if (preferences.usePreciseAngle.get() && !dataState.locked().get() && !dataState.allAdvancementsDataState().allAdvancementsModeEnabled().get()) {
+			if (!dataState.boatDataState().enteringBoat().get() && dataState.boatDataState().reducingModulo360().get())
+				actionExecutor.executeImmediately(new ToggleMod360IndicatorAction(dataState));
 			actionExecutor.executeImmediately(new ToggleEnteringBoatAction(dataState));
```

Comment:
Each call to actionExecutor.executeImmediately will save the state for undo/redo. This means that if both actions are executed here, to undo the hotkey press the user needs to undo twice, and undoing once will cause the program to end up in a weird state.

To fix this just call executeImmediately with both actions as parameters, see PlayerPositionInputHandler.onNewDetailedPlayerPositionInputted(...) for an example.

Reply:
Fixed in commit [ea505c8](https://github.com/Ninjabrain1/Ninjabrain-Bot/pull/91/commits/ea505c8a0963ef1c36b6befffa3baf7a00fa7575). Some logic was able to be simplified as a result of the changes in commit [71f1a5f](https://github.com/Ninjabrain1/Ninjabrain-Bot/pull/91/commits/71f1a5fd2b5b2a5530b060c0feee51aab4f7f26d).